[Console] Avoid constructor promotion in 5.4 branch

tstarling opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| License       | MIT

The 5.4 branch only requires PHP 7.2 so should not use constructor promotion.

For downstream https://phabricator.wikimedia.org/T301344

tstarling committed:
Avoid constructor promotion in 5.4 branch

The 5.4 branch only requires PHP 7.2 so should not use constructor promotion.

For downstream https://phabricator.wikimedia.org/T301344

tstarling committed:
Fix fabbot style nitpicks

tstarling commented:
AppVeyor failed with a timeout, probably not my fault.

michaljusiega commented:
And you can't still to use native attributes in PHP >=7.2 and < 8.0 so that's changes are unnecessary IMO.

tstarling commented:
Attributes parse as comments in earlier versions of PHP, so that's not a problem, although the trailing comma in the parameter list is a syntax error which I will fix shortly.

tstarling committed:
Remove trailing comma in parameter list

Remove trailing comma in parameter list, which is also a syntax error before PHP 8.0.

nicolas-grekas commented:
Thanks for the link to the issue that triggered this PR.
This class is one among many others that isn't supposed to be loaded when PHP < 8 is used.
The proper solution is to add a rule in the linter to skip those classes.

tstarling commented:
There is a rule to skip the linter [here](https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/vendor/+/3f686eddfdf3348628226bf74fac2cca184f34e0/composer.json#153) but that is not the only place we do linting. There is only one file out of 4570 PHP files in our vendor directory that fails to parse with PHP 7.2, and this is it. So this is the most general place to fix it.

As an alternative, you could exclude AsCommand.php from the 5.4 packages using archive.exclude in composer.json.

nicolas-grekas commented:
> There is only one file out of 4570 PHP files in our vendor directory that fails to parse with PHP 7.2, and this is it.

That's only because you don't use Symfony enough :stuck_out_tongue_closed_eyes: 

We periodically have similar reports about other files, but our policy here is clear: the classes are expected to be loaded via the autoloader, and conditional loading is a thing we rely on. Other tools should adapt.

> you could exclude AsCommand.php from the 5.4 packages using archive.exclude in composer.json

That wouldn't work as 5.4 (and eg this attribute) can be used on PHP 8.

Thanks for your understanding, I'm now closing the PR since there is no chance this might be accepted.
